index.js

```diff
@@ -30,6 +33,23 @@ module.exports = async function(obj) {
     throw new Error('serviceName required');
   }
 
+
+  const queryTasks = async function(serviceName) {
+    const opts = {
+      filters: `{"service": [ "${serviceName}" ] }`
+    }
+
+    const tasks = await client.listTasks(opts);
+    if (tasks[0].Status.State !== 'running' && tasks[0].Status.State !== 'failed') {
```

Comment:
why just checking the first task?

Reply:
also - if a task as failed in the past, does this account for that?

Reply:
I think I read that you can check an UpdateStatus.  Grab all the tasks that are Updating and then make sure all of those task ids get to running

Reply:
another option is to listen for docker events  - see docker-watch